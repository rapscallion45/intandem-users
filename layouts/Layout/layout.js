import CssBaseline from '@mui/material/CssBaseline';
import Box from '@mui/material/Box';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Footer from '../../components/footer';
import Logo from '../../components/logo';
import Link from '../../components/link';

const Index = function Index({ children }) {
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        minHeight: '100vh',
      }}
    >
      <CssBaseline />
      <Box sx={{ flexGrow: 1, maxHeight: 80 }}>
        <AppBar position="sticky">
          <Toolbar variant="dense">
            <Link href="/" sx={{ mr: 5 }}>
              <Logo />
            </Link>
            <Typography variant="h6" color="inherit" component="div">
              Intandem Users
            </Typography>
          </Toolbar>
        </AppBar>
      </Box>
      {children}
      <Footer />
    </Box>
  );
};

export default Index;
