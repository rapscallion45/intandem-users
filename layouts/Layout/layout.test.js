import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { render } from '@testing-library/react';
import router from 'next/router';
import Layout from './layout';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      route: '/user',
      pathname: '',
      query: { id: 7 },
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    };
  },
}));

describe('Layout', () => {
  it('Renders correctly enzyme', () => {
    /* Arrange */
    const TestChild = function TestChild() {
      return <div>Test Child</div>;
    };
    const useRouter = jest.spyOn(router, 'useRouter');
    useRouter.mockImplementation(() => ({
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
    }));

    /* Act */
    const wrapper = shallow(
      <Layout>
        <TestChild />
      </Layout>
    );

    /* Assert */
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Elements', () => {
    it('Should render header with correct title text', () => {
      /* Arrange */
      const TestChild = function TestChild() {
        return <div>Test Child</div>;
      };
      const titleText = 'Intandem Users';
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = render(
        <Layout>
          <TestChild />
        </Layout>
      );

      /* Assert */
      expect(wrapper.getByText(titleText)).toBeInTheDocument();
    });

    it('Should render link to home page', () => {
      /* Arrange */
      const TestChild = function TestChild() {
        return <div>Test Child</div>;
      };
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = render(
        <Layout>
          <TestChild />
        </Layout>
      );

      /* Assert */
      expect(wrapper.getByRole('link', { name: 'logo-image' })).toBeInTheDocument();
    });

    it('Should render passed children components', () => {
      /* Arrange */
      const TestChild = function TestChild() {
        return <div>Test Child</div>;
      };
      const useRouter = jest.spyOn(router, 'useRouter');
      useRouter.mockImplementation(() => ({
        route: '/',
        pathname: '',
        query: '',
        asPath: '',
        push: jest.fn(),
        events: {
          on: jest.fn(),
          off: jest.fn(),
        },
        beforePopState: jest.fn(() => null),
        prefetch: jest.fn(() => null),
      }));

      /* Act */
      const wrapper = render(
        <Layout>
          <TestChild />
        </Layout>
      );

      /* Assert */
      expect(wrapper.getByText('Test Child')).toBeInTheDocument();
    });
  });
});
