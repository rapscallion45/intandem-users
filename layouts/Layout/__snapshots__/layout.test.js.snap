// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Renders correctly enzyme 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "minHeight": "100vh",
    }
  }
>
  <CssBaseline />
  <ForwardRef(Box)
    sx={
      Object {
        "flexGrow": 1,
        "maxHeight": 80,
      }
    }
  >
    <ForwardRef(AppBar)
      position="sticky"
    >
      <ForwardRef(Toolbar)
        variant="dense"
      >
        <ForwardRef
          href="/"
          sx={
            Object {
              "mr": 5,
            }
          }
        >
          <Logo />
        </ForwardRef>
        <ForwardRef(Typography)
          color="inherit"
          component="div"
          variant="h6"
        >
          Intandem Users
        </ForwardRef(Typography)>
      </ForwardRef(Toolbar)>
    </ForwardRef(AppBar)>
  </ForwardRef(Box)>
  <TestChild />
  <Footer />
</ForwardRef(Box)>
`;
